File: ruby/0226-invert-binary-tree.rb
def invert_tree(root)
  return root if root.nil?
  return root if root.left.nil? && root.right.nil?

  root.left, root.right = root.right, root.left
  invert_tree(root.left)
  invert_tree(root.right)
  root
end

File: rust/0226-invert-binary-tree.rs
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn invert_tree(root: Option<Rc<RefCell<TreeNode>>>) -> Option<Rc<RefCell<TreeNode>>> {
        root.map(|node| {
            {
              let mut node_ref = node.borrow_mut();
              let left = node_ref.left.take();
              let right = node_ref.right.take();
              node_ref.right = Solution::invert_tree(left);
              node_ref.left = Solution::invert_tree(right);
            }
            node
        })
    }
}
